osquery/tables/system/posix/smbios_utils.cpp

```diff
@@ -309,6 +397,158 @@ void genSMBIOSMemoryDevices(size_t index,
   results.push_back(std::move(r));
 }
 
+void genSMBIOSMemoryArrays(size_t index,
+                           const SMBStructHeader* hdr,
+                           uint8_t* address,
+                           size_t size,
+                           QueryData& results) {
+  if (hdr->type != kSMBIOSTypeMemoryArray || size < 0x12) {
+    return;
+  }
+
+  Row r;
+  r["handle"] = dmiWordToHexStr(address, 0x02);
+
+  auto location = kSMBIOSMemoryArrayLocationTable.find(address[0x04]);
+  if (location != kSMBIOSMemoryArrayLocationTable.end()) {
+    r["location"] = location->second;
+  }
+
+  auto use = kSMBIOSMemoryArrayUseTable.find(address[0x05]);
+  if (use != kSMBIOSMemoryArrayUseTable.end()) {
+    r["use"] = use->second;
+  }
+
+  auto errCorrection =
+      kSMBIOSMemoryArrayErrorCorrectionTypesTable.find(address[0x06]);
+  if (errCorrection != kSMBIOSMemoryArrayErrorCorrectionTypesTable.end()) {
+    r["memory_error_correction"] = errCorrection->second;
+  }
+
+  auto cap = dmiToDWord(address, 0x07);
+  // SMBIOS returns capacity in KB or bytes, but we want a more human
+  // friendly GB.
+  r["max_capacity"] = (cap >= 0x80000000)
+                          ? INTEGER(dmiToQWord(address, 0x0F) / 1073741824)
+                          : INTEGER(cap / 1048576);
+
+  auto errHandle = dmiToWord(address, 0x0B);
+  if (errHandle != 0xFFFE) {
+    r["memory_error_info_handle"] =
+        (errHandle == 0xFFFF) ? "No Errors" : toHexStr(errHandle);
```

Comment:
nit: Would it be better to not define "memory_error_info_handle" in the absence of errors, as opposed to the string "No Errors"? I'm just thinking in terms of what would be easier to filter by when querying, like `<> ''` is easier and less error prone than `<> 'No Errors'`.

Reply:
I think you're right, it absolutely would be.  Will change.